days/day-03/solutions/day03.zig

```diff
@@ -0,0 +1,66 @@
+const std = @import("std");
+const mem = std.mem;
+const io = std.io;
+const ArrayList = std.ArrayList;
+
+const MAX_FILE_SIZE = 20_000;
+
+const Map = struct {
+    bytes: []u8,
+    line_width: usize,
+    map_height: usize,
+
+    pub inline fn init(bytes: []u8) Map {
+        var line_width: usize = 0;
+        while (line_width < bytes.len) : (line_width += 1) {
+            switch(bytes[line_width]) {
+                '\n', '\r' => break,
+                else => continue
+            }
+        }
+
+        line_width += 1;
+
+        return Map {
+            .bytes = bytes,
+            .line_width = line_width,  // TODO: account for arbitrary line ending format
+            .map_height = @divFloor(bytes.len, line_width)
+        };
+    } 
+
+    pub inline fn inspectPath(self: Map, rigth: usize, down: usize) usize {
+        var x_pos: usize = 0;
+        var y_pos: usize = 0;
+        var closed: u32 = 0;
+
+        while (y_pos < self.map_height) {
+            var n_skipped = @divFloor(x_pos, self.line_width - 1);
+            var real_x = (x_pos + n_skipped) % self.line_width;
+            var real_y = y_pos * self.line_width;
+
+            closed += @boolToInt(self.bytes[real_x + real_y] == '#');
```

Comment:
Is the `@` signifying compile-time in this language?

Reply:
@Arxcis Comptime has no extra syntax in  Zig other than "comptime" @ mean builtin (in the compiler) function. Its in other words language features that does a given operation (like casting)

To run a expression on compile time you simply say `comptime *expression*`. See:
- https://kristoff.it/blog/what-is-zig-comptime/
- https://ziglang.org/documentation/master/#comptime

Reply:
:smile: